source/connectors/kafka-sink-postprocessors.txt

```diff
@@ -9,10 +9,821 @@ Kafka Sink Connector Post-Processors
+       | You can set this explicitly with the following configuration: ``writemodel.strategy=com.mongodb.kafka.connect.sink.writemodel.strategy.DeleteOneDefaultStrategy``
+
+   * - UpdateOneTimestampsStrategy
+     - | Add ``_insertedTS`` (inserted timestamp) and ``_modifiedTS`` (modified timestamp) fields into documents.
+       | Set the following configuration: ``writemodel.strategy=com.mongodb.kafka.connect.sink.writemodel.strategy.UpdateOneTimestampsStrategy``
+
+       .. seealso:: Example of usage in :ref:`writemodel-strategy-timestamps`.
+
+.. note::
+
+   Future versions of the MongoDB Kafka Connector will allow developers to
+   specify a user-created, custom strategy in the configuration settings.
+
+.. _writemodel-strategy-business-key:
+
+WriteModel Strategy: Business Keys
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A business key is a value comprised of data within the sink record that
+identifies it as a unique document. This example shows how to define
+a business key using the data contained by multiple fields in the record as
+well as instruct the post processor to generate *BSON ObjectIds* for
+inserts, but not for updates.
+
+The following steps are required to configure this strategy:
+
+#. Create a **unique index** that corresponds to your business key in your
+   target MongoDB collection.
+#. In the connector configuration, specify the ``PartialValueStrategy``
+   as the id strategy to identify the fields that belong to the business
+   key.
+#. In the connector configuration, specify the
+   ``ReplaceOneBusinessKeyStrategy`` writemodel strategy.
+
+Suppose the parking usage data was being tracked for city blocks, defined
+by distinct values for ``street`` and ``avenue``. An example message
+contains the following:
+
+.. code-block:: json
+
+   {
+     "street": "50th Street",
+     "avenue": "Broadway",
+     "parking_spaces": {
+       "capacity": 407,
+       "occupied": 352
+     }
+   }
+
+In order to implement the strategy, we first create a unique index on the
+``street`` and ``avenue`` fields in the MongoDB shell:
+
+.. code-block:: shell
+
+   db.collection.createIndex({ "street": 1, "avenue": 1}, { unique: true })
+
+Next, we specify the ``PartialValueStrategy`` strategy and fields to
+include in the business key and specify the
+``ReplaceOneBusinessKeyStrategy`` writemodel strategy in the configuration
+file:
+
+.. code-block:: properties
+
+   document.id.strategy=com.mongodb.kafka.connect.sink.processor.id.strategy.PartialValueStrategy
+   value.projection.list=street,avenue
+   value.projection.type=whitelist
+   writemodel.strategy=com.mongodb.kafka.connect.sink.writemodel.strategy.ReplaceOneBusinessKeyStrategy"
+
+The sample data inserted into the collection contains the following:
+
+.. code-block:: json
+   :emphasize-lines: 2
+
+   {
+     "_id": ObjectId('5db0b81199b7fbcc89631d06'),
+     "street": "50th Street",
+     "avenue": "Broadway",
+     "parking_spaces": {
```

Comment:
just a note that I was surprised to see parking_spaces still included with the whitelist of street,avenue